modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -609,18 +609,27 @@ def _read(cls, path, engine, columns, **kwargs):
         )
         from modin.pandas.io import PQ_INDEX_REGEX
 
-        if isinstance(path, str) and os.path.isdir(path):
+        if isinstance(path, str):
+            if os.path.isdir(path):
+                path_generator = os.walk(path)
+            else:
+                storage_options = kwargs.get("storage_options")
+                if storage_options is not None:
+                    fs, fs_path = url_to_fs(path, **storage_options)
+                else:
+                    fs, fs_path = url_to_fs(path)
+                path_generator = fs.walk(fs_path)
             partitioned_columns = set()
             # We do a tree walk of the path directory because partitioned
             # parquet directories have a unique column at each directory level.
             # Thus, we can use os.walk(), which does a dfs search, to walk
             # through the different columns that the data is partitioned on
-            for (_, dir_names, files) in os.walk(path):
+            for (_, dir_names, files) in path_generator:
                 if dir_names:
                     partitioned_columns.add(dir_names[0].split("=")[0])
                 if files:
                     # Metadata files, git files, .DSStore
-                    if files[0][0] == ".":
+                    if len(files[0]) > 0 and files[0][0] == ".":
```

Comment:
Do you know why we only check the first file in the directory? i.e. `files[0]`?

Reply:
If I understand the current logic correctly, it’s “if there’s any directory within path where the first file [and apparently the files don’t come in any guaranteed order] starts with ‘.’, then stop looking for partitioned columns”. I think is buggy in a few ways, e.g. see #4637. For now could you just add a TODO to fix the logic here?